compiler/src/vm/builtin_functions.rs

```diff
@@ -165,148 +183,139 @@ impl Vm {
         })
     }
 
-    fn use_(&mut self, args: Vec<Value>) -> Result<Value, String> {
+    fn use_asset(&mut self, db: &Database, args: Vec<Value>) -> Result<Value, String> {
+        let (current_path, target) = Self::parse_current_path_and_target(args)?;
+        let target = UseAssetTarget::parse(&target)?;
+        let input = target.resolve_asset(&current_path)?;
+        let content = db
+            .get_input(input.clone())
+            .ok_or_else(|| format!("Couldn't import file '{}'.", input))?;
+        Ok(Value::list(
+            content
+                .iter()
+                .map(|byte| Value::Int(*byte as u64))
+                .collect_vec(),
+        ))
+    }
+
+    fn use_local_module(&mut self, db: &Database, args: Vec<Value>) -> Result<(), String> {
+        let (current_path, target) = Self::parse_current_path_and_target(args)?;
+        let target = UseAssetTarget::parse(&target)?;
+        let possible_inputs = target.resolve_local_module(&current_path)?;
+        let input = 'find_existing_input: {
+            for input in possible_inputs {
+                if db.get_input(input.clone()).is_some() {
+                    break 'find_existing_input input;
+                }
+            }
+            return Err("couldn't import module".to_string());
+        };
```

Comment:
Isn't something like the following possible?

```suggestion
        let input = possible_inputs.iter()
            .filter_map(|it| db.get_input(it.clone()))
            .next()
            .or_or_else("couldn't import module".to_string())?;
```